clang/lib/AST/ItaniumMangle.cpp

```diff
@@ -1518,9 +1518,10 @@ void CXXNameMangler::mangleUnqualifiedName(GlobalDecl GD,
     // <lambda-sig> ::= <template-param-decl>* <parameter-type>+
     //     # Parameter types or 'v' for 'void'.
     if (const CXXRecordDecl *Record = dyn_cast<CXXRecordDecl>(TD)) {
-      if (Record->isLambda() && (Record->getLambdaManglingNumber() ||
-                                 Context.getDiscriminatorOverride()(
-                                     Context.getASTContext(), Record))) {
+      if (Record->isLambda() &&
+          (Record->getLambdaManglingNumber() ||
+           Context.getDiscriminatorOverride()(Context.getASTContext(), Record)
+               .getValueOr(0))) {
```

Comment:
Could you please explain this change?

Reply:
`getDiscriminatorOverride()` returns an `Optional`. If that optional doesn't hold a value, this treats it as if it held `0`. The predicate then tests if the value is zero (whether because the optional held no value or it held the value zero explicitly).

I believe `0` is special as a mangling number; we assert that it's not zero elsewhere (like line 1966/1968 below).

Reply:
Yep, thats basically it :)  '0' gets mangled as 'not a lambda' and just a nameless struct.  That said, i think there is a little bit of an issue with the logic here that 'works', but could be better expressed.